test/routes/action.test.js

```diff
@@ -0,0 +1,64 @@
+// let actionService
+require('../../server/services/actionService')
+const mockActionServiceResult = { success: true }
+
+function createMocks () {
+  jest.mock('../../server/services/actionService', () => ({
+    performAction: jest.fn(() => mockActionServiceResult)
+  }))
+  /* actionService = require('../../server/services/actionService')
+  actionService.performAction = jest.fn(() => (actionServiceMockResult)) */
+}
+
+function extractSessionCookie (response) {
+  const setCookie = response.headers['set-cookie']
+  return (setCookie && setCookie[0]) ? setCookie[0].split(';')[0] : ''
+}
+
+function getRedirectOptions (response) {
+  const cookie = extractSessionCookie(response)
+  return {
+    method: 'GET',
+    headers: { cookie },
+    url: response.headers.location
+  }
+}
+
+describe('Action test', () => {
+  let createServer
+  let server
+
+  beforeAll(async () => {
```

Comment:
The function was incorrectly marked as an `async` function in the project these were based on, might be worth removing the async from test functions that do not contain `await`